pkg/pty/pty_linux.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2015-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package pty provides basic pty support.
+// It implments much of exec.Command
+// but the Start() function starts two goroutines that relay the
+// data for Stdin, Stdout, and Stdout such that proper kernel pty
+// processing is done. We did not simply embed an exec.Command
+// as we can no guarantee that we can implement all aspects of it
+// for all time to come.
+package pty
+
+import (
+	"os"
+	"os/exec"
+	"strconv"
+	"syscall"
+	"unsafe"
+
+	"github.com/u-root/u-root/pkg/termios"
+)
+
+// pty support. We used to import github.com/kr/pty but what we need is not that complex.
+// Thanks to keith rarick for these functions.
+func New(cmd string, args ...string) (*Pty, error) {
+	tty, err := termios.New()
+	if err != nil {
+		return nil, err
+	}
+	restorer, err := tty.Get()
+	if err != nil {
+		return nil, err
+	}
+	ptm, err := os.OpenFile("/dev/ptmx", os.O_RDWR, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	err = ptsunlock(ptm)
+	if err != nil {
+		return nil, err
+	}
+
+	sname, err := ptsname(ptm)
+	if err != nil {
+		return nil, err
+	}
+
+	// It can take a non-zero time for a pts to appear, it seems.
+	// Ten tries is reported to be far more than enough.
+	for i := 0; i < 10; i++ {
+		_, err := os.Stat(sname)
+		if err == nil {
```

Comment:
I could swear the error4 I got was kinda weird and was not IsNotExist. The tty code is a wonder to behold, so I'd rather just go with 'try to open 10 times, give up'.